src/app/app.component.ts

```diff
@@ -273,6 +275,41 @@ export class AppComponent {
     }
 
     this.initPerimeterX();
+
+    document.addEventListener("fullscreenchange", () => {
+      this.toggleSkipToMainContent()
+    }, false);
+  
+    document.addEventListener("mozfullscreenchange", () => {
+      this.toggleSkipToMainContent()
+    }, false);
+  
+    document.addEventListener("webkitfullscreenchange",() => {
+      this.toggleSkipToMainContent()
+    }, false);
+  
+    document.addEventListener("msfullscreenchange", () => {
+      this.toggleSkipToMainContent()
+    }, false);
+  }
+
+  private toggleSkipToMainContent() {
+    if (this.isFullscreen) {
+        this.addSkipToMainContent()
+    } else {
+      this.removeSkipToMainContent()
+    }
+    this.isFullscreen = !this.isFullscreen
+  }
+
+  private removeSkipToMainContent() {
+    let skipToMainContentButton = document.getElementById("skip-main-content-button")
+    skipToMainContentButton.tabIndex = -1
+  }
+
+  private addSkipToMainContent() {
+    let skipToMainContentButton = document.getElementById("skip-main-content-button")
+    skipToMainContentButton.tabIndex = 1
```

Comment:
So this adding/removing the "skip to main content" control... Is this still necessary now that we have changed the element we give to the `requestFullScreen`? If I understand correctly the `skip-main-content-button` lives outside of the `main` element.

Reply:
Not necessary for chrome where the `requestFullScreen` works well with the element, but this doesn't work for Firefox. The dom still contains all body with Firefox.

Reply:
hmm interesting. I wonder why that is the case for firefox. Seems like we could save ourselves a lot of code if we didn't have to account for this weird edge case (I know we hide/show the nav/footer in this way too for that reason). It might be worth spending some more time trying to get to the bottom of that? This isn't happening on the jstor side of things right? I wonder if it has to do with the us feeding the full screen a `main` element as opposed to a div or something?

Reply:
I tired div before I switched to `main`. It didn't work so it's not the problem of `main`.

Reply:
Hmm interesting. I now remember that Openseadragon removes everything else from the DOM when it goes into full screen mode using the default controls, which creates an entirely different set of complexity. https://openseadragon.github.io/

Emulating this may be worth discussing but it may be pretty far out of scope.

Reply:
hmm so I just tested our jstor iiif viewer on firefox and noticed it suffers from the same problem. So it's likely a bigger issue. Interesting how openseadragon employs a similar fix. I am cool with going with these work arounds for now. Resolving this comment